src/Pages/Detail/Detail.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import API from '../../config';
+import Goods from './DetailSections/Goods';
+import Categories from './DetailSections/Categories';
+import Popup from '../../Components/ShoppingModal/Popup';
+import Login from '../Login/Login';
+import './Detail.scss';
+
+class Detail extends React.Component {
+  state = {
+    goods: {},
+    isModalOpen: false,
+    isLogin: false,
+    content: {},
+  };
+
+  handleModal = (e, amount) => {
+    const { name } = e.target;
+    const { isLogin, goods } = this.state;
+    const { id, korean_name, price } = goods;
+
+    const order = { id, korean_name, amount, price };
+
+    if (!isLogin) {
+      this.addOrder(order);
+      if (name === 'buy') {
+        this.props.history.push('/payment');
+        return;
+      }
+      this.setState({ isModalOpen: true });
+      return;
+    }
+
+    if (name === 'buy' || name === 'cart') {
+      this.setState({ isModalOpen: true });
+      return;
+    }
+
+    this.setState({ isModalOpen: false });
+  };
+
+  addOrder = data => {
+    const token = localStorage.getItem('access_token');
+    const resource = `/orders/cart`;
+
+    fetch(API.detail + resource, {
```

Comment:
```suggestion
    return fetch(API.detail + resource, {
```
promise를 활용할 것이기 때문에 fetch 함수의 리턴 값 자체를 리턴해주는게 좋겠습니다

Reply:
넵 프로미스 리턴하여 사용하겠습니다.